### DISCUSSION TITLE
Proposal: podcast:storefront & JSON spec

### DISCUSSION THREAD

**Discussion by nathangathright** [Podcastindex-org/podcast-namespace#644] [Spec Proposal]
### Purpose
Merch has long been a way for listeners to show their support for their favorite podcasters and musicians. As [Spotify](https://newsroom.spotify.com/2023-10-16/shop-find-all-artist-merch-hub-spotify-app-shopify/) and [Amazon Music](https://press.aboutamazon.com/2021/3/amazon-music-launches-new-shopping-experience-making-it-easier-for-fans-to-find-merch-from-their-favorite-artists) incorporate product storefronts into their listening apps, I wanted to explore what it would take to bring that innovation to the podcasting ecosystem. As I’m imagining it, feeds could link to a JSON file of products with links to complete the purchase via the web.

### Parent
`<channel>`

### Count
Single

### Attributes
- **url:** (required) A URL pointing to a JSON file.

### Example

```xml
<podcast:storefront url="https://example.com/12345678/merch.json"/>
```

---

# JSON Format

This is the initial spec for a JSON format that can be referenced in an RSS feed using the `<podcast:storefront/>` tag. This file can reside on any publicly accessible URL. This file should be served with a Content-Type of `application/json` and a [Access-Control-Allow-Origin header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin) set to `*`.

## JSON Keys
- **version:** (required - string) The version number of the format.
- **storefront:** (required - array)
  - **title:** (required - string)
  - **description:** (required - string)
  - **link:** (required - string)
  - **image:** (required - string)
## Example
```json
{
  "version": "1.0.0",
  "storefront": [
    {
      "title": "ATP Monochrome Shirt",
      "description": "The ATP logo shirt in several fun colors.",
      "link": "https://cottonbureau.com/p/ZREXJY/shirt/atp-monochrome",
      "image": "https://cottonbureau.com/p/ZREXJY/shirt/atp-monochrome/image.jpg"
    },
    {
      "title": "ATP Hat",
      "description": "Let the proud ATP logo protect your head from the harmful rays of the sun.",
      "link": "https://cottonbureau.com/p/DEQQPB/hat/atp-hat",
      "image": "https://cottonbureau.com/p/DEQQPB/hat/atp-hat/image.jpg"
    },
    {
      "title": "ATP Enamel Pin",
      "description": "The classic ATP logo pin, now with a locking pin back.",
      "link": "https://cottonbureau.com/p/GXUGXR/pin/atp-enamel-pin",
      "image": "https://cottonbureau.com/p/GXUGXR/pin/atp-enamel-pin/image.jpg"
    }
  ]
}

```

**Comment by theDanielJLewis** [comment]
I like the idea. What about in the (likely) absence of the correctly formatted JSON? Could we simply link to the storefront?

I recommend allowing multiple uses of the tag and with text, allowing for the multiple storefronts podcasters might have. Plus, a `provider` (or similar) attribute could help for icons, branding, or even an abbreviated version of multiple buttons.

For example:

```xml

<podcast:storefront url="https://stickermule.com/mystore" provider="Sticker Mule">Make your stuff awesome with our stickers</podcast:storefront>
<podcast:storefront url="https://teepublic.com/mystore" provider="TeePublic">Be instantly sexy in our tees!</podcast:storefront>
<podcast:storefront url="https://amazon.com/mystore"provider="Amazon">Buy my sock collection</podcast:storefront>
```

Then, a podcast app could display a button for each storefront and use the full CTA text, or it could read the "provider" field to match to simple icon-only buttons under something like "Podcast swag."

That idea of grouping them makes me wonder if there should be a parent element wrapping them and that could include the title of the group of links and maybe an optional description.

**Reply by nathangathright** [reply]
In my opinion, the existing `funding` tag is sufficient to cover the majority of what these changes might achieve.
* Podcasters can include multiple per channel/item
* Apps can identify the provider by parsing the hostname from the URL
* Apps can display links from known storefront domains with icons in a dedicated storefront area.

If correctly-formatted JSON is too burdensome, then this tag shouldn’t exist at all. But I think many hosts could facilitate an "add your merch" UI that results in a storefront on their host-generated websites and a JSON file in their feeds.

**Comment by samsethi** [comment]
I am goin g to start playing with this proposal for TrueFans.  We will offer podcasters a merch store in their admin dashboard.  We will also include the price as an attribute.

**Comment by matthewruzzi** [comment]
Could we use Schema.org JSON-LD for this?
https://schema.org/ItemList
https://schema.org/ListItem
https://schema.org/Product

ChatGPT generated example:
```json
{
  "@context": "https://schema.org",
  "@type": "ItemList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@type": "Product",
        "name": "ATP Monochrome Shirt",
        "description": "The ATP logo shirt in several fun colors.",
        "url": "https://cottonbureau.com/p/ZREXJY/shirt/atp-monochrome",
        "image": "https://cottonbureau.com/p/ZREXJY/shirt/atp-monochrome/image.jpg"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@type": "Product",
        "name": "ATP Hat",
        "description": "Let the proud ATP logo protect your head from the harmful rays of the sun.",
        "url": "https://cottonbureau.com/p/DEQQPB/hat/atp-hat",
        "image": "https://cottonbureau.com/p/DEQQPB/hat/atp-hat/image.jpg"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@type": "Product",
        "name": "ATP Enamel Pin",
        "description": "The classic ATP logo pin, now with a locking pin back.",
        "url": "https://cottonbureau.com/p/GXUGXR/pin/atp-enamel-pin",
        "image": "https://cottonbureau.com/p/GXUGXR/pin/atp-enamel-pin/image.jpg"
      }
    }
  ]
}
```

**Reply by samsethi** [reply]
I like the idea of structured data https://developers.google.com/search/docs/appearance/structured-data/product 

But this complexity has to be hidden from podcasters. The value is for Google to index the product.  The biggest challenges are not displaying the products but payments and delivery/returns.  I have built big ecommerce platforms before. 

This is why I think links to external storefronts might be easier for now.  So do we use the funding tag like Daniel suggested above or do we create a seperate tag for storefront links like Nathan suggested?  

The  recent Kelsy Brothers ClubHouse [announcement](https://podnews.net/press-release/kelce-clubhouse) I find interesting as it is full Amazon underneath with a branded skin on top and a collection of products for the Kelsey Brothers. 

I could use that URL https://www.amazon.com/stores/page/DC3A251C-7D33-49D8-9277-EBD9E348247D inside either the fiunding tag or the storefront tag. 

At the moment on TrueFans we have a donation icon when the funding tag is active. If it is was as storefront I would prefer to show a shopping cart app.  So podcasters might want both a funding url and storefront url.  Yes both urls could live inside the fudning tag.  

Maybe the storefront tag is only used if the host has a proper store front option to upload product images, set prices etc and that storefront then appears inside the podcast app.  TrueFans wants to offer podcasters a storefront option.  Most podcasters will only have a few products to sell.  

The more I think about it, links to a third party store with payment and delivery options is probably a better option.

<img width="1056" height="140" alt="TrueFans___Shows___Buzzcast" src="https://github.com/user-attachments/assets/9ef8f9a4-d1bc-4bee-a122-5a77cdb4a78a" />

**Reply by nathangathright** [reply]
Podcasters should be shown UI to upload images and fill out the text strings. They never need be aware if there’s a JSON file at all. If the host provides the podcaster a website, they could display these products on a new Merch page, and link that page in the `podcast:funding` tag to maximize reach in apps that don’t support the `podcast:storefront` tag yet. Hosts could even offer a white-label integration with Printful or Fourthwall to produce and ship the merch if they so chose. 

If we’re not interested in showing all the products directly in podcast apps, then the funding tag is the way to go. It already supports a **Node Vale** where podcasters can include a link label in case they need to disambiguate multiple:

```
<podcast:funding url="https://patreon.com/...">Support us on Patreon</podcast:funding>
<podcast:funding url="https://amazon.com/stores/page/...">Buy our merch!</podcast:funding>
```

TrueFans could even create a pseudo-Linktree out of these funding links. (Maybe pull each link’s `og:image` `og:title` and `og:description` metadata for a rich preview card potentially?

**Reply by nathangathright** [reply]
@matthewruzzi I have no objection to JSON-LD except that the other JSON specs in the namespace don’t also support it. Happy for the app developers who’re actually willing to implement this to decide that. Feel free to make a new discussion proposing an updated spec for all the existing JSON namespace specs.

**Comment by samsethi** [comment]
Hello 

TrueFans has implemented a very simple Merchant Store link based on Daniel/Nathan suggestions. 

In their admin dashboard creators can add one or several stores with links and descriptions

<img width="1353" height="575" alt="TrueFans___Dashboard" src="https://github.com/user-attachments/assets/2bf40e61-5539-4a19-952d-a2a7f6cbc376" />

On their podcast page you get a cart icon ,with a tooltip and link out. 

<img width="1440" height="749" alt="Fullscreen_26_02_2026__20_40" src="https://github.com/user-attachments/assets/0be9c8ee-a218-4918-8b8a-cf797ae5c75d" />


The next step we will allow creators to upload images, descriptions and links out for fulfillment.  This is exactly what Spotify and YouTube do today.  We will display these products on the merch tab. 

The final step we will allow podcasts we host to take the payment for products directly on TrueFans using either Stripe or Bitcoin from their wallet

We have already added the podcast:storefront tag to the RSS feed. 


<img width="1181" height="427" alt="TrueFans___Dashboard" src="https://github.com/user-attachments/assets/74bc8f48-e710-4216-b8f0-60cdcb59aef5" />

<img width="1000" height="249" alt="TrueFans___Dashboard" src="https://github.com/user-attachments/assets/8c4c4749-b1ea-48fc-b525-396f3918895f" />
